Arrangement_on_surface_2/include/CGAL/Curved_kernel_via_analysis_2/gfx/Curve_renderer_internals.h

```diff
@@ -914,9 +907,9 @@ void get_precached_poly(int var, const NT& key, int /* level */, Poly_1& poly)
 
     if(not_cached||not_found) {
         poly = Poly_1(::boost::make_transform_iterator(coeffs->begin(),
-                            boost::bind2nd(std::ptr_fun(binded_eval), key1)),
+                          std::function<NT(const Poly_1&)>([&key1](const Poly_1& poly){ return evaluate(poly, key1); })),
                       ::boost::make_transform_iterator(coeffs->end(),
-                            boost::bind2nd(std::ptr_fun(binded_eval), key1)));
+                          std::function<NT(const Poly_1&)>([&key1](const Poly_1& poly){ return evaluate(poly, key1); })));
```

Comment:
Why use `std::function` here? In your version, there are two lambdas, of different types but identical implementation, and you store them in a `std::function` to get a common type.

I think you should use only one lambda, directly:
```c++
auto transform_fn = [&key1](const Poly_1& poly){ return evaluate(poly, key1); };
poly = Poly_1(::boost::make_transform_iterator(coeffs->begin(), transform_fn),
              ::boost::make_transform_iterator(coeffs->end(),   transform_fn));
```